scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/ParquetAvroFileOperations.java

```diff
@@ -141,7 +146,9 @@ protected FileIO.Sink<ValueT> createSink() {
   @SuppressWarnings("unchecked")
   @Override
   public Coder<ValueT> getCoder() {
-    return (AvroCoder<ValueT>) AvroCoder.of(getSchema());
+    return recordClass == null
+        ? (AvroCoder<ValueT>) AvroCoder.of(getSchema())
+        : AvroCoder.of(recordClass, true);
```

Comment:
How did this work before? Codec for GenericRecord was used for SpecificRecords?

Reply:
I made a bug in beam https://github.com/apache/beam/pull/29518. In current version it will use `SpecificData`. Prefer explicit `AvroCoder.reflect`